graspy/align/seedless_procrustes.py

```diff
@@ -0,0 +1,328 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import ot
+import numpy as np
+from sklearn.utils import check_array
+
+from .base import BaseAlign
+from .sign_flips import SignFlips
+from .orthogonal_procrustes import OrthogonalProcrustes
+
+
+class SeedlessProcrustes(BaseAlign):
+    """
+    Implements an algorithm that matches two datasets using an orthogonal
+    matrix. Unlike OrthogonalProcrustes, this does not use a matching between
+    entries. It can even be used in the settings when the two datasets do not
+    have the same number of entries.
+
+    In graph setting, it is used to align the embeddings of two different
+    graphs, when it requires some simultaneous inference task, for example,
+    inside of the test for the equivalence of the latent distributions.
+
+    In essence, it aims to simultaneously obtain a, not necessarily 1-to-1,
+    correspondance between the vertices of the two data sets, and the
+    orthogonal matrix alignment matrix. It does via a sequence of alternating
+    steps, similarly to a regular expectation-maximization procedures. The step
+    of obtaining an assignment ("E-step") is done by solving an optimal
+    transport problem via Sinkhorn algorithm, whereas obtaining an orthogonal
+    alignment matrix ("M-step") is done via regular orthogonal procurstes.
+
+    Parameters
+    ----------
+        optimal_transport_lambda : float (default=0.1), optional
+            Regularization term of the Sinkhorn optimal transport algorithm.
+
+        optimal_transport_eps : float (default=0.01), optional
+            Tolerance parameter for the each Sinkhorn optimal transport
+            algorithm. I.e. tolerance for each "E-step".
+
+        optimal_transport_num_reps : int (default=1000), optional
+            Number of repetitions in each iteration of the iterative optimal
+            transport problem. I.e. maximum number of repetitions in each
+            "E-step".
+
+        iterative_eps : float (default=0), optional
+            Tolerance of the iterative optimal transport problem. I.e.
+            tolerance of the whole "EM" algorithm. Zero by default, which means
+            that the algorithm performs all `iterative_num_reps` number of
+            steps.
+
+        iterative_num_reps : int (default=100), optional
+            Number of reps in each iteration of the iterative optimal transport
+            problem. I.e. maxumum number of total iterations the whole "EM"
+            algorithm.
```

Comment:
lets discuss naming of kwargs (for this and the `optimal_transport` stuff). these seem a bit long and i have a hard time understanding what all they do, and I have the benefit of having talked to people about this stuff.

Reply:
I would just say that longer is better if only because `ot` is as meaningless as randomly picking 2 letters from any alphabet and stringing them together - for those of us without context, that is

Reply:
@bdpedigo , i'm not sure if you were present for that part of conversation with carey, but i think that having iterative_eps is somewhat useless, and may even be misleading, since the for most problems objective functions does not go to zero. so, i offer to remove it. this leaves us with:
```
        optimal_transport_lambda=0.1,
        optimal_transport_eps=0.01,
        optimal_transport_num_reps=1000,
        iterative_num_reps=100,
        init="2d",
```
for simplicity sake, i would not mind omitting optimal_transport where unnecessary (so have it just `lambd` and `eps`), but personally that feels more confusing to me personally. i'm totally happy to follow any other convention you prefer.